### PR TITLE
[Imaging-268] Add list of TIFF files and example survey application

### PR DESCRIPTION
**Pull request by gwlucastrig**
This pull request provides a summary of the various configurations for TIFF files in the Commons Imaging test resources. The summary is stored in a file named README.txt at the top-level directory of test files.

I have also included the example applications that survey the test folder and list the files it finds.  The example applications can also produce a .csv file suitable for reading into Excel or relational databases.

### CODE REVIEW

**Review by kinow — COMMENTED** [review]
Couple files that I will need the IDE to go through the new code.

@gwlucastrig thanks for taking care to create the JIRA issue and reference here in the PR title. Later once you have your ASF ID, we should be able to include the changelog entry in `changes.xml` in this PR too :clap: 

I've been adding your changes with my user, but now you should be able to write something like

`<action issue="IMAGING-268" dev="you-asf-id" type="add">....description...</action>`

If you generate the site with `mvn clean site`, you should see under the Changes section your new entry. And the release-manager will perform some commands during the release. One of the commands reads the changes too and produces the RELEASE-NOTES.txt that is included in jar's and in the other archives and site :+1:

**Comment by gwlucastrig** [comment]
The example code is now ready for final review and incorporation into the code base.

Also, I tried the mvn site command you suggested, but it failed with an error.  I will research it later, but for now, here's the text:

 An Ant BuildException has occured: Execute failed: java.io.IOException: Cannot run program "svn": CreateProcess error=2, The system cannot find the file specified
[ERROR] around Ant part ...<exec executable="svn">... @ 4:26 in c:\Work\commons-imaging\target\antrun\build-main.xml

File: src/test/java/org/apache/commons/imaging/examples/tiff/SurveyTiffFile.java [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+        boolean imageDataInStrips = directory.imageDataInStrips();
+        if (imageDataInStrips) {
+            final TiffField rowsPerStripField
+                = directory.findField(TiffTagConstants.TIFF_TAG_ROWS_PER_STRIP);
+            rowsPerStrip = Integer.MAX_VALUE;
+            if (null != rowsPerStripField) {
+                rowsPerStrip = rowsPerStripField.getIntValue();
+            } else {
+                final TiffField imageHeight = directory.findField(TiffTagConstants.TIFF_TAG_IMAGE_LENGTH);
+                /**
```
Comment (kinow) [inline src/test/java/org/apache/commons/imaging/examples/tiff/SurveyTiffFile.java:118]: `/*` as this is just a general comment.

File: src/test/java/org/apache/commons/imaging/examples/tiff/SurveyTiffFolder.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            if (i > 0) {
+                String ext = args[1].substring(i);
+                if (".csv".equalsIgnoreCase(ext)) {
+                    csv = true;
+                }
+            }
+            File reportFile = new File(args[1]);
+            try (FileOutputStream fos = new FileOutputStream(reportFile);
+                BufferedOutputStream bos = new BufferedOutputStream(fos);
+                PrintStream ps = new PrintStream(bos, true, "UTF-8");) {
```
Comment (kinow) [inline src/test/java/org/apache/commons/imaging/examples/tiff/SurveyTiffFolder.java:86]: Unnecessary `;`

**Comment by kinow** [comment]
Few minor comments, ready to merge after feedback :+1:

**Comment by gwlucastrig** [comment]
It am fortunate to have such a thorough reviewer.  Thanks for the notes.  I've submitted the updates.

**Comment by kinow** [comment]
Thanks @gwlucastrig !! This should definitely help users (and devs like myself!) to get more familiar with the TIFF parser and related code.  Merging.

**Review by kinow — APPROVED** [review]

**Pull request closed without merging**